api/src/main/java/jakarta/resource/spi/ManagedConnection.java

```diff
@@ -224,7 +223,6 @@ void removeConnectionEventListener(
    *  @return     ManagedConnectionMetaData instance
    *
    *  @throws     ResourceException     generic exception if operation fails
-   *  @throws     NotSupportedException if the operation is not supported
```

Comment:
Was this removed by mistake?

Reply:
No it's not in the method signature so javadoc complains

Reply:
A JavaDoc complaint doesn't justify altering the programming contract.  Currently, with Java EE, an implementation that interacts with a ManagedConnection can rely on receiving a NotSupportedException to signal that the resource adapter does not support the operation.  If we remove the `@throws NotSupportedException` from the contract that is defined by the JavaDoc, then it becomes no longer valid for an implementation to rely upon this signal, which becomes a breaking behavior change between Java EE and Jakarta.  Instead, the problem should be addressed by officially adding `NotSupportedException` to the throws clause of the method, which would be a non-breaking change because `NotSupportedException` is a subclass of `ResourceException` which is already declared to be raised by the method.

Reply:
Will that then require a signature test change?

Reply:
Not sure.

Reply:
@alwin-joseph Could you check the older signatures? If this is in the API, but isn't in the signatures, that is probably a mistake.

Reply:
signature for jakartaee9 :https://github.com/eclipse-ee4j/jakartaee-tck/blob/7d1ff5365a621928f2feac75c8d0378dea0071fe/src/com/sun/ts/tests/signaturetest/signature-repository/jakarta.resource.sig_2.0_se8#L383

signature for jakartaee8 : https://github.com/eclipse-ee4j/jakartaee-tck/blob/4687f0a79e3ce4f25b8505a7281b603de6455ae9/src/com/sun/ts/tests/signaturetest/signature-repository/javax.resource.sig_1.7_se8#L823

It does not look like the api had the "throws  NotSupportedException" in the removeConnectionEventListener, getLocalTransaction & getXAResource **method signature** for JakartaEE8. 

The  "*  @throws     NotSupportedException if the operation is not supported" in the method comments will not be reflected in the signature files.

Reply:
Adding the throws looks like it would then fail the signature test. So what's consensus. 

1. Revert the change and see if we can get the javadoc generation to work
2. add the throws 
3. change the javadoc by deleting the @throws?

Reply:
If the signature cannot be fixed, I think option 4 would be acceptable.  Maybe something along the lines of:
```
*  @throws     ResourceException if the operation fails. If the operation is not supported then jakarta.resource.NotSupportedException, which is a subclass of ResourceException, must be raised.
```

Reply:
I reinstated the throws and added the import as suggested by @pzygielo